src/neighbor.cpp

```diff
@@ -2396,9 +2415,38 @@ void Neighbor::modify_params(int narg, char **arg)
 	  ex_mol_intra[nex_mol] = 0;
         nex_mol++;
         iarg += 3;
-	
+
+      } else if (strcmp(arg[iarg+1],"custom/match") == 0 ||
+		 strcmp(arg[iarg+1],"custom/clash") == 0) {
+        if (iarg+4 > narg) error->all(FLERR,"Illegal neigh_modify command");
+        int is_double;
+        int index = atom->find_custom(arg[iarg+2],is_double) + 1;
+        if (index == 0)
+          error->all(FLERR,"Neigh_modify exclude custom requires predefined property/atom");
+        else if (is_double)
+          error->all(FLERR,"Neigh_modify exclude custom requires integer-valued property/atom");
+
+        if (nex_ivec == maxex_ivec) {
+          maxex_ivec += EXDELTA;
+          memory->grow(ex_ivec_group,maxex_ivec,"neigh:ex_ivec_group");
+          if (lmp->kokkos)
+//            grow_ex_ivec_intra_kokkos();
+            error->all(FLERR,"Neigh_modify exclude custom not implemented for package kokkos");
+          else
+            memory->grow(ex_ivec_index,maxex_ivec,"neigh:ex_ivec_index");
+        }
+        ex_ivec_group[nex_ivec] = group->find(arg[iarg+3]);
+        if (ex_ivec_group[nex_ivec] == -1)
```

Comment:
This is a very hackish approach. Rather than depending on positive/negative, i would suggest to add CUSTOM_MATCH and CUSTOM_CLASH constants to the enumerator at the top (after TEMPLATE) and use a separate list to indicate the custom exclusion choice. this leaves room for other custom exclusion patterns and makes the code more readable and particularly lifts the requirement to store the property index with an offset of 1 added.
as for looking up custom indices. i have to revert my previous assessment. the use seems ok. however, a more paranoid implementation would simply store the custom property names (as a copy!) and look them up, check, and index them in Neighbor::init()

Reply:
I agree. This was inspired by the molecule/intra and molecule/inter exclusions and I just replaced 1/0 by positive/negative because we now need to store an index. I should have added some comments, though. I'm going to implement your suggestion.